arlo-client/src/components/utilities.test.ts

```diff
@@ -56,4 +57,59 @@ describe('utilities.ts', () => {
       expect(testNumber(50)(100)).resolves.toBe('Must be smaller than 50')
     })
   })
+
+  describe('poll', () => {
+    it('iterates', async () => {
+      const j = (function* idMaker() {
+        var index = 0
+        while (true) yield index++
+      })()
+      let result = ''
+      const condition = async () => j.next().value > 2
+      const callback = async () => (result = 'over five')
+      const error = async () => (result = 'an error')
+      await poll(condition, callback, error)
+      await wait(() => {
+        expect(result).toBe('over five')
+      })
+    })
+
+    it('times out', async () => {
+      const j = (function* idMaker() {
+        var index = 1
+        while (true) yield index++
+      })()
+      let result = ''
+      const condition = async () => j.next().value < 1
+      const callback = async () => (result = 'over five')
+      const error = async () => (result = 'an error')
+      await poll(condition, callback, error, 50, 10)
+      await wait(() => {
+        expect(result).toBe('an error')
+      })
+    })
+
+    it('times out with spies', async () => {
+      const startDate: number = Date.now()
+      const lateDate: number = startDate + 130000
+      const dateSpy = jest
+        .spyOn(Date, 'now')
+        .mockReturnValueOnce(startDate)
+        .mockReturnValueOnce(lateDate)
+      const j = (function* idMaker() {
+        var index = 1
+        while (true) yield index++
+      })()
+      let result = ''
+      const condition = async () => j.next().value < 1
```

Comment:
```suggestion
      const condition = async () => false
```

We never want this condition to resolve to `true`, so I don't think we need the whole `idMaker` thing.